formats/onenote.ts

```diff
@@ -0,0 +1,345 @@
+	async generateLoginCode(): Promise<string> {
+		const requestBody = new URLSearchParams({
+			client_id: GRAPH_CLIENT_ID,
+			scope: GRAPH_SCOPES.join(' '),
+		});
+
+		// Using requestUrl in order to prevent CORS issues
+		const tokenResponse = await requestUrl(
+			`https://login.microsoftonline.com/common/oauth2/v2.0/devicecode?${requestBody.toString()}`
+		);
+
+		const deviceCodeData: DeviceCode = await tokenResponse.json;
+		await this.pollForAccessToken(deviceCodeData);
+		return deviceCodeData.user_code;
+	}
+
+	async pollForAccessToken(deviceCodeRequest: DeviceCode) {
+		const requestBody = new URLSearchParams({
+			grant_type: 'urn:ietf:params:oauth:grant-type:device_code',
+			client_id: GRAPH_CLIENT_ID,
+			device_code: deviceCodeRequest.device_code,
+		});
+		let intervalId = setInterval(async () => {
+			try {
+				const tokenResponse = await requestUrl({
+					method: 'POST',
+					url: 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
+					contentType: 'application/x-www-form-urlencoded',
+					body: requestBody.toString(),
+				});
+				const tokenData: TokenResponse = tokenResponse.json;
+				await this.signIn(tokenData.access_token);
+				clearInterval(intervalId);
+			}
+			catch (e) {
+				console.log(e);
+			}
+		}, deviceCodeRequest.interval * 1000); // Convert seconds into miliseconds
+	}
+	
+	async signIn(accessToken: string) {
+		this.requestParams = {
+			headers: {
+				Authorization: `Bearer ${accessToken}`,
+			},
+		};
+		await this.showSectionPickerUI();
+	}
+
+	async showSectionPickerUI() {
+		const response = await fetch(
+			'https://graph.microsoft.com/v1.0/me/onenote/notebooks?$expand=sections($select=id,displayName)&$select=id,displayName&$orderby=createdDateTime',
+			this.requestParams
+		);
+		const data = await response.json();
+		const notebooks: Notebook[] = data.value;
+
+		// Replace the sign in area to declutter the UI
+		const contentArea: HTMLDivElement = this.modal.contentEl.querySelector('.contentArea')!;
+		contentArea.innerHTML = '';
+		contentArea.createEl('h3', {
+			text: 'Choose what to import',
+			cls: 'modal-title',
+		});
+		contentArea.createEl('hr');
+
+		notebooks.forEach((notebook) => {
+			contentArea.createEl('h5', {
+				text: notebook.displayName!,
+				cls: 'modal-title',
+			});
+			let sections: OnenoteSection[] = notebook.sections!;
+			// All notebooks have sections but not all notebooks have section groups
+			let sectionGroups: SectionGroup[] | undefined | null = notebook?.sectionGroups;
+
+			this.createSectionList(sections);
+			sectionGroups?.forEach((sectionGroup) => {
+				this.modal.contentEl.createEl('h6', {
+					text: sectionGroup.displayName!,
+				});
+				this.createSectionList(sectionGroup.sections!);
+			});
+		});
+	}
+
+	createSectionList(sections: OnenoteSection[]) {
+		const contentArea: HTMLDivElement = this.modal.contentEl.querySelector('.contentArea')!;
+
+		sections?.forEach((section) => {
+			let label = contentArea.createEl('label');
```

Comment:
have you considered using Obsidian `Setting` for this? that would ensure the UI is consistent with rest of Obsidian